src/aind_metadata_mapper/fib/session.py

```diff
@@ -0,0 +1,295 @@
+            experimenter_full_name=settings.experimenter_full_name,
+            session_start_time=settings.session_start_time,
+            session_end_time=settings.session_end_time,
+            session_type=settings.session_type,
+            rig_id=settings.rig_id,
+            subject_id=settings.subject_id,
+            iacuc_protocol=settings.iacuc_protocol,
+            notes=settings.notes,
+            data_streams=data_streams,
+            mouse_platform_name=settings.mouse_platform_name,
+            active_mouse_platform=settings.active_mouse_platform,
+        )
+
+        return session
+
+    def run_job(self) -> JobResponse:
+        """Run the complete ETL job and return a JobResponse.
+
+        This method orchestrates the full ETL process:
+        1. Extracts metadata from files and settings
+        2. Transforms the data into a valid Session object
+        3. Saves the session metadata to the specified output location
+        4. Verifies the output file was written correctly
+
+        Returns:
+            JobResponse: Object containing status code, message, and optional
+                data. Status codes:
+                - 200: Success
+                - 406: Validation errors
+                - 500: File writing errors
+
+        Notes:
+            If a custom output filename is specified in job_settings, it will
+            be used instead of the default filename.
+        """
+        extracted = self._extract()
+        transformed = self._transform(extracted)
+
+        # If a custom filename is specified, save it directly
+        if self.job_settings.output_filename:
+            output_path = (
+                self.job_settings.output_directory
+                / self.job_settings.output_filename
+            )
+            try:
+                with open(output_path, "w") as f:
+                    f.write(transformed.model_dump_json(indent=2))
+                if verify_output_file(
+                    self.job_settings.output_directory,
+                    self.job_settings.output_filename,
+                ):
+                    return JobResponse(
+                        status_code=200,
+                        message=f"Successfully wrote file to {output_path}",
+                        data=None,
+                    )
+                else:
+                    return JobResponse(
+                        status_code=500,
+                        message=f"File verification failed for {output_path}",
+                        data=None,
+                    )
+            except Exception as e:
+                return JobResponse(
+                    status_code=500,
+                    message=f"Error writing to {output_path}: {str(e)}",
+                    data=None,
+                )
+
+        # Otherwise use the default write_standard_file method
+        return self._load(transformed, self.job_settings.output_directory)
+
+
+def update_job_settings_with_times(
```

Comment:
It seems like this method should either be added to the utils package or be added as a class method.

Reply:
I removed this function entirely.